queries.lisp

```diff
@@ -450,7 +450,7 @@
 ;;; Instances
 
 (defmethod instance ((client client))
-  (decode-instance (query client "/api/v1/instance")))
+  (decode-instance (query client "/api/v2/instance")))
```

Comment:
I'm a bit concerned about backwards compatibility with older servers here. We might need to query the available API version and switch based on that. Is that even possible to do?

Reply:
Right concerns! I can check if it is possible to just call the most recent version (v2) and fall back to v1 if not. The only issue i can see is the compatibility of the object returned by the two endpoints, let me test if such compatibility exists!

Reply:
I have added a workaround for fallback mechanism, i have tested it with an instance running mastodon version 4.2: no issues fortunately! What do you think about it? Is it OK?

Reply:
Hmm, how about creating subclasses of `client` instead, and using `change-class` to select the version when the client is first initialised? That way we only need to query once and can encode the version differences into the method specialisers instead.

Reply:
Nice idea @Shinmera ! I like it a lot! Any idea for the child class: something like `client-version-1`, or `client-legacy`? very open to suggestion, i am not good with names! :smile:

Reply:
I guess if you want to be really fancy you could make a new `org.shirakumo.tooter.client.v2` package that uses the old client package and just shadows client, then local-nickname the package to v2 in `org.shirakumo.tooter` so you could refer to it as `v2:client`.

Reply:
I have added `client-legacy` (but i would rather use a more specific name, like `client-v1` or even `client-4.2`, so we could accommodate other versions like: `client-4.3` when, 4.4 will be released and so on, what do you think. BTW, regardless of the name, have I implemented correctly your idea?

Reply:
I think having `client` as the non-specific base class, `client-v1`, and `client-v2` would be good, mirroring the API endpoints mastodon exposes.

Reply:
Fine with me! Ia going to change the name.

Reply:
Though client version selection could be automated. In `authorize` and `shared-initialize` if the necessary access-token is provided you can perform the call to figure out the API version supported, and then `change-class` into the appropriate version subclass.

Reply:
> Though client version selection could be automated. In `authorize` and `shared-initialize` if the necessary access-token is provided you can perform the call to figure out the API version supported, and then `change-class` into the appropriate version subclass.

Sure! Not sure how to to it exactly but i guess i will find a way to figure out!

Reply:
OK @Shinmera, i modified the method `authorize` to change class if the instance type can be guessed (i.e. if a call to `instance` API endpoint can be performed), let me know if these changes are good.

C.

Reply:
> I guess if you want to be really fancy you could make a new `org.shirakumo.tooter.client.v2` package that uses the old client package and just shadows client, then local-nickname the package to v2 in `org.shirakumo.tooter` so you could refer to it as `v2:client`.

I added a new package to accommodate the client version, i am not really sure if i implemented correctly what you asked, but i think the code improved someway. :sweat_smile: